File: questPresentation/src/main/java/ru/nekit/android/qls/lockScreen/content/AbstractLockScreenContentMediator.java
package ru.nekit.android.qls.lockScreen.content;

import android.support.annotation.NonNull;

import ru.nekit.android.qls.lockScreen.TransitionChoreograph;

public abstract class AbstractLockScreenContentMediator {

    @NonNull
    protected TransitionChoreograph mTransitionChoreograph;

    public void setTransitionChoreograph(@NonNull TransitionChoreograph choreograph) {
        mTransitionChoreograph = choreograph;
    }

    @NonNull
    public abstract ILockScreenContentContainerViewHolder getContentContainerViewHolder();

    public abstract void deactivate();

    public abstract void detachView();

    public abstract void attachView();

}

File: questPresentation/src/main/java/ru/nekit/android/qls/lockScreen/content/common/AbstractLockScreenContentMediator.kt
package ru.nekit.android.qls.lockScreen.content.common

import io.reactivex.disposables.CompositeDisposable
import ru.nekit.android.qls.quest.providers.IQuestContextProvider

abstract class AbstractLockScreenContentMediator : IQuestContextProvider {

    override var disposable: CompositeDisposable = CompositeDisposable()

    abstract val viewHolder: ILockScreenContentViewHolder

    abstract fun deactivate()

    abstract fun detachView()

    abstract fun attachView()

}